### PR TITLE
0-shel subject and audit

### PR DESCRIPTION
**Pull request by MSilva95**

### CODE REVIEW

File: subjects/0-shell/audit/README.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+###### Can you confirm that the displayed message of the project is exactly the same as the computer terminal?
+
+##### Try to open the project and run the command `echo something else` (without double quotes). Do the same in your computer terminal.
+
+###### Can you confirm that the displayed message of the project is exactly the same as the computer terminal?
+
+##### Try to open the project and run the command `pwd`.
+
+###### Can you confirm that the interpreter displayed the current path?
```
Comment (LEEDASILVA) [inline subjects/0-shell/audit/README.md:33]: maybe you can add some questions like
` ##### Try to open the project and create a parent folder with two child folders using the command "mkdir". Then enter the parent folder and do "pwd" . `

`###### Can you confirm that the interpreter displayed the current path?`

**Review by LEEDASILVA — CHANGES_REQUESTED** [review]

File: subjects/0-shell/README.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
+## 0-shell
+
+### Objective
+
+The objective of this project is for you to create a simple [shell](https://en.wikipedia.org/wiki/Shell_script).
```
Comment (xpetit) [inline subjects/0-shell/README.md:5]: https://en.wikipedia.org/wiki/Unix_shell looks like a more appropriate link

File: subjects/0-shell/README.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+Through the `0-shell` you will get to the core of the `Unix` system and explore an important part of this system’s API witch is the process creation and synchronization.
+Executing a command inside a shell implies creating a new process, which execution and final state will be monitored by its parents process. This set of functions will be the key to success for your project.
+
+For this project you will only have to create a simple `Unix` interpreter where you can run some of the most known commands. For this part of the project, no advanced functions, pipes or redirection will be asked, you can add them if you like.
```
Comment (xpetit) [inline subjects/0-shell/README.md:10]: "shell", "Unix shell" or "command-line interpreter" but not "Unix interpreter". Unix can't be interpreted

File: subjects/0-shell/README.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+### Instructions
+
+- You must program a mini `Unix` interpreter.
+- This interpreter must display at least a simple `$` and wait until you type a command line witch will be validated by pressing enter.
+- The `$` will be shown again only once the command has been completely executed.
+- The command lines are simple, you will not have pipes, redirection or any other advanced functions.
+- You must manage the errors, by displaying a message adapted to the error output.
+- You must implement the following commands that will behave exactly like the originals:
```
Comment (xpetit) [inline subjects/0-shell/README.md:19]: There are no originals unfortunately.
So maybe we can advise to target a simple and widespread thing, like BusyBox.

File: subjects/0-shell/README.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+  - ls
+  - pwd
+  - cat
+  - cp
+  - rm
+  - mv
+  - mkdir
+  - exit
+- You must manage the program interruption `Ctrl + D`.
+- The project has to be written in a compiled language like (C, Rust, go or other), no semi compiled language like (Pearl and others) are allowed.
```
Comment (xpetit) [inline subjects/0-shell/README.md:31]: "go" => "Go"
"semi compiled" => "interpreted"
"Pearl" => "Perl"

File: subjects/0-shell/job-control/README.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+Job control refers to the ability to selectively stop (suspend) the execution of processes and continue (resume) their execution at a later point.
+
+In `job control`, you will have to implement the following [builtins](https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Job-Control-Builtins):
+
+- jobs
+- bg
+- fg
+- kill
+- disown
```
Comment (xpetit) [inline subjects/0-shell/job-control/README.md:15]: To follow BusyBox I would replace `disown` with `nohup` or remove this builtin entirely as it is less important.

File: subjects/0-shell/scripting/README.md [hunk 7]
```diff
@@ -0,0 +1,75 @@
+## scripting
+
+### Objectives
+
+You must follow the same [principles](https://public.01-edu.org/subjects/0-shell/) as the first subject.
+
+In `scripting` you must find a way to make your `0-shell` run `bash` scripts.
```
Comment (xpetit) [inline subjects/0-shell/scripting/README.md:7]: Bash is too complex and has too many advanced structures to be taken as a example.
`ash` instead ? (integrated in BusyBox and in Debian as `dash`)

File: subjects/0-shell/scripting/README.md [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+```
+for ((i = 0 ; i < 100 ; i++)); do
+  echo $i
+done
+```
+
+```
+STR="HELLO WORLD!"
+echo ${STR,}   #=> "hELLO WORLD!" (lowercase 1st letter)
```
Comment (xpetit) [inline subjects/0-shell/scripting/README.md:25]: This is Bash and not POSIX compliant so I would remove these string manipulations.

**Review by xpetit — CHANGES_REQUESTED** [review]

**Review by xpetit — APPROVED** [review]

**Review by LEEDASILVA — APPROVED** [review]

**Pull request merged**